streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSource.java

```diff
@@ -86,12 +90,16 @@ public void init(final ProcessorContext context) {
             metrics = (StreamsMetricsImpl) context.metrics();
             droppedRecordsSensor = droppedRecordsSensorOrSkippedRecordsSensor(Thread.currentThread().getName(), context.taskId().toString(), metrics);
             if (queryableName != null) {
-                store = (TimestampedKeyValueStore<K, V>) context.getStateStore(queryableName);
+                store = (TimestampedSerializedKeyValueStore<K, V>) context.getStateStore(queryableName);
```

Comment:
@vvcephei  Yeah, I definitely noticed too. I've just had a thought. Should we do a refactoring for the state store hierarchy first before we further proceed with this PR?  That way, we can avoid a lot of the wacky wrangling we are doing here.

Reply:
Nooooo :) 

I've tried tackling that particular dragon a couple of times. It's an absolutely enormous undertaking. Not too "difficult" per se, but designing for compatibility is a bit hairy. The real problem is just the massive diff that results.

My plan is to finish up the implementation of KIP-478 and then take it on, since 478 will make it a bit safer to make massive internal changes like that.

It's a good thought, but I'd rather let both efforts proceed in parallel. Especially since the code I proposed above, wile mildly offensive, is still pretty safe.